src/v/kafka/server/protocol.cc

```diff
@@ -125,13 +129,38 @@ config::broker_authn_method get_authn_method(const net::connection& conn) {
       && config.enable_sasl()) {
         return config::broker_authn_method::sasl;
     }
-    // mtls_identity is currently predicated on having mapping rules
-    if (conn.get_principal_mapping().has_value()) {
-        return config::broker_authn_method::mtls_identity;
-    }
     return config::broker_authn_method::none;
 }
 
+ss::future<security::tls::mtls_state> get_mtls_principal_state(
+  const security::tls::principal_mapper& pm, net::connection& conn) {
+    using namespace std::chrono_literals;
+    return ss::with_timeout(
+             model::timeout_clock::now() + 5s, conn.get_distinguished_name())
+      .then([&pm](std::optional<ss::session_dn> dn) {
+          ss::sstring p;
+          if (!dn.has_value()) {
+              vlog(klog.info, "failed to fetch distinguished name");
+              return security::tls::mtls_state{std::move(p)};
+          }
+          auto principal = pm.apply(dn->subject);
+          if (!principal) {
+              vlog(
+                klog.info,
+                "failed to extract principal from distinguished name: {}",
+                dn->subject);
+              return security::tls::mtls_state{std::move(p)};
+          }
```

Comment:
can we extract a principal and have it be an empty string for some reason? it looks like an empty string is a signal to the caller than something went wrong, but below we'd log it at debug level rather than info. perhaps we should handle that case too?

Reply:
An empty principal i the anonymous user, as far as I can tell, which seems like a valid case, although I suppose it might be confusing, if something is misconfigured and debug logging isn't on.

Reply:
that sounds like reasonable behavior. do you know what kafka is doing if an empty-string principal is extracted?